Src/CSharpier.Cli/ServerFormatter.cs

```diff
@@ -1,39 +1,46 @@
-using System.Net.NetworkInformation;
-using CSharpier.Proto;
-using Grpc.Core;
-using Microsoft.Extensions.Logging;
-
 namespace CSharpier.Cli;
 
 using System.IO.Abstractions;
+using System.Net;
+using System.Net.NetworkInformation;
 using CSharpier.Cli.Options;
+using Microsoft.AspNetCore.Builder;
+using Microsoft.AspNetCore.Hosting;
+using Microsoft.Extensions.Logging;
 
-public static class GrpcFormatter
+public static class ServerFormatter
 {
-    public static Task<int> StartServer(
+    public static async Task<int> StartServer(
         int? port,
         ConsoleLogger logger,
         string? actualConfigPath,
         CancellationToken cancellationToken
     )
     {
         var thePort = port ?? FindFreePort();
-        var server = new Server
-        {
-            Services =
-            {
-                CSharpierService.BindService(
-                    new CSharpierServiceImplementation(actualConfigPath, logger)
-                )
-            },
-            Ports = { new ServerPort("localhost", thePort, ServerCredentials.Insecure) }
-        };
-        server.Start();
-
+        var builder = WebApplication.CreateBuilder();
+        builder
+            .WebHost
+            .ConfigureKestrel(
+                (_, serverOptions) =>
+                {
+                    serverOptions.Listen(IPAddress.Any, thePort);
```

Comment:
So, you can actually use [unix sockets](https://learn.microsoft.com/en-us/dotnet/api/microsoft.aspnetcore.server.kestrel.core.kestrelserveroptions.listenunixsocket?view=aspnetcore-8.0) (Linux, possibly Mac) and [named pipes](https://learn.microsoft.com/en-us/dotnet/api/microsoft.aspnetcore.server.kestrel.core.kestrelserveroptions.listennamedpipe?view=aspnetcore-8.0) (Windows) to avoid having to chose a port.

Reply:
That will make it much easier if you ever plan on having multiple editors share an instance of csharpier.

Reply:
The proto models are no longer needed, I probably forgot to delete them.

I took a look at both a bit. One concern was windows/linux/mac + java/js/c# is a lot of combinations of things to figure out.

Also with named pipes/unix sockets, I assume I still need to come up with a unique name for each instance of CSharpier that runs. If you are running rider with 0.27 and vscode with 0.27 right now that is two instances of csharpier. I assume they would be unhappy all trying to share the same socket/pipe.

Reply:
Yes. With UNIX sockets, you'd be using (probably):
```bash
"${XDG_RUNTIME_DIR}/csharpier/${version}.sock"
```

Which (in my machine) expanded to:
```
/run/user/1000/csharpier/0.27.2.sock
```

I'm not sure what the conventions are for windows named pipes, but I assume a similar thing is possible.